### ISSUE TITLE
auto-type config for groups,global shortcut for locking database

### ISSUE THREAD

**Issue by renzhamin** [keepassxreboot/keepassxc#6440]
##  Global shortcut for locking database
I am wondering why this is not available in a program like keepassxc or maybe there's already a way to do this?
Currently I'm using xdotool to send keystrokes but it works about 50% of the time
Note : I run no bs like system tray

## Default Auto-Type window titles
I have become a huge fan of the auto-type feature,I have noticed that regex can be used,but I havent found this mentioned in the docs
### Observation
Most sites are in this format **"prefix.domain.{\*}.{com|org|net|any_suffix}"**
And the title for these sites is generated as **"prefix.domain.suffix"** where prefix and suffix can be empty string,so generally
The first or the second element of title.split('.') will match domain
The login window titles are almost always matched by **\*Domain\*** entry, this matching has worked for me 100% of the time
#### Example
Site : https://github.com/login
Automatically generated title : **github.com**
Matching window title for login url : **\*Github\***

You may say that just use the page tittle accordingly to match it,but in practical none of us are coming to keepass with just a couple of passwords,we got tons and tons of them,so the title is created from the previous method we used to store passwords before keepass or from the browser-extension

EDIT : The easier way to go about this would be to support exporting csv with auto-type configuration included,then we can run some script on title column and autotype column to modify all the entries at once

### Implementation
So it boils down to,
Enabling an option to use default window titles by using the output of a regex performed on title/url for **database/group**
It doesnt really has to be a regex,because title.split()[0] will match the domain part for more than 60% of the sites

I'm really interested in what Auto-Type will become and with this feature it might to able to truly replace the need of browser extension.Maybe not 100% but its a step forward

**Comment by droidmonkey** [comment]
Not likely we will add a global shortcut to lock databases. We offer many security options already to auto lock your database on various triggers. 

Your second request is already implemented (I think) in develop. It was very hard to follow your request.

**Comment by gr523** [comment]
>Not likely we will add a global shortcut to lock databases. We offer many security options already to auto lock your database on various triggers.

Great. The built-in triggers are super useful  and the global shortcut is also needed specially if you are working on a public environment

For the auto-type feature,
Basically we have a autotype section for all individual entry config,if there is something similar on group config then its infinitely more easier to configure autotypes
Secondly,I know we can use the placeholder but it would be more flexible if we could use some kind of parsing on these placeholders and then use the parsed string to batch configure window tittles

**Comment by gr523** [comment]
For those who would like to batch configure autotype window title described in the stated scheme,here's what I did(dirty workaround)
1.Export database using cli
2.Add window title association on every entry to match \*\<tittle\>\*
3.Import database
4.Run the python script

```
keepassxc-cli export Passwords.kdbx > test.xml

sed -i 's/<\/AutoType>/<Association><Window>*\&lt;tittle\&gt;*<\/Window><KeystrokeSequence\/><\/Association>\n<\/AutoType>\n/g' test.xml

keepassxc-cli import test.xml testdb.kdbx

pip install --user pykeepass
curl -s https://raw.githubusercontent.com/gr523/Codes/master/Python/keepassAutoTypeConfigure.py > keepassAutoTypeConfigure.py

python3 keepassAutoTypeConfigure.py
```
done,now open testdb.kdbx and see the results
There will be some broken tittles if you have sites like prefix.domain.... where prefix is not empty and unusual(not www)
You can go around this by appending those prefix's in the prefix set in the python script
